ZL-Asica/KumoAuth
# feat: OpenAPI reference auto generated by zod, Scalar for interactive

Code:
1. 4xx 5xx responses are now { { error: message }, statusCode }
2. 2xx responses are now { { data_json }, 2xx }

New Structure:
1. index.ts: entry changed to OpenAPIHono() from Hono()
1. lib/db: All query.
2. lib/helper: for zod json/response/response structures.

Test:
1. register test.

Fix:
1. register logic to check database query status from db feedback.

Dependencies:
1. Use [zod-openai](https://hono.dev/examples/zod-openapi) middleware for doc generating. (at /doc)
4. Use [scalar](https://github.com/scalar/scalar/blob/main/packages/hono-api-reference/README.md) for interactive API reference. (at /reference)
5. Add vite and vite-tsconfig-path as dev-dependencies for alias resolving.
